src/main/java/search/BinarySearch.java

```diff
@@ -9,25 +9,22 @@ public class BinarySearch {
      * Array must be sorted prior to search.
      * Efficiency: O(log n)
      * @param sortedArr array of elements to be searched. Must be sorted in order for algorithm to function properly.
-     * @param value value to search for within the array.
+     * @param searchValue value to search for within the array.
      * @return the index of the value if it is found within the array, -1 if not found.
      */
-    public int binarySearch(int[] sortedArr, int value) {
-        int left = 0;
-        int right = sortedArr.length - 1;
-        while(left <= right) {
-            int mid = (left + right) / 2;
-            if(sortedArr[mid] == value) {
-                return mid;
-            }
-            if(sortedArr[mid] < value) {
-                left = mid + 1;
-            }
-            else {
-                right = mid - 1;
+    public Integer binarySearch(int[] sortedInput, int searchValue) {
+        int leftIndex = 0, rightIndex = sortedInput.length - 1;
+        while(leftIndex <= rightIndex) {
+            final int midIndex =  (leftIndex + rightIndex) / 2;
+            if(sortedInput[midIndex] == searchValue) {
+                return midIndex;
+            } else if(sortedInput[midIndex] > searchValue) {
+                rightIndex = midIndex - 1;
+            } else {
+                leftIndex = midIndex + 1;
             }
         }
-        return -1;
```

Comment:
Personally I'd just keep the `-1` rather than using the boxed `Integer` with a null. Or if you want a more modern pattern, something like `bool TryBinarySearch(int[] arr, int val, out index)`.

Reply:
Sir, this is a Java